Extension/test/integrationTests/MockDebugger/debugAdapterDescriptorFactory.ts

```diff
@@ -0,0 +1,47 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All Rights Reserved.
+ * See 'LICENSE' in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import * as vscode from "vscode";
+import * as path from 'path';
+
+// import * as util from '../src/common' <- DO NOT USE. Also do not use anything with relative paths, it will break during hijacking in test/debug.integration.test.ts
+
+abstract class DebugAdapterDescriptorFactoryWithContext implements vscode.DebugAdapterDescriptorFactory {
+    protected readonly context: vscode.ExtensionContext;
+
+    constructor(context: vscode.ExtensionContext) {
+        this.context = context;
+    }
+
+    abstract createDebugAdapterDescriptor(session: vscode.DebugSession, executable: vscode.DebugAdapterExecutable | undefined): vscode.ProviderResult<vscode.DebugAdapterDescriptor>;
+}
+
+export class CppdbgDebugAdapterDescriptorFactory extends DebugAdapterDescriptorFactoryWithContext {
```

Comment:
Does this support interfacing so that we can create an interface of `ITestDebugAdapter` and then we can make sure in the production case that we don't have that interface but we do in the test case?